spec/classes/hyperglass_agent_spec.rb

```diff
@@ -2,17 +2,131 @@
 
 describe 'hyperglass::agent' do
   let :node do
-    'rspec.puppet.com'
+    'rspec.example.com'
   end
 
   on_supported_os.each do |os, facts|
     context "on #{os} " do
       let :facts do
-        facts
+        facts.merge(
+          python3_version: '3.6.8'
+        )
       end
 
       context 'with all defaults' do
         it { is_expected.to compile.with_all_deps }
+
+        it {
+          is_expected.to contain_file('/opt/hyperglass').with(
+            {
+              'ensure' => 'directory',
+              'owner'  => 'root',
+              'group'  => 'root',
+              'mode'   => '0755',
+            }
+          )
+        }
+
+        it { is_expected.to contain_class('python').with_version('3').with_dev('present') }
+
+        it { is_expected.to contain_package('gcc').with_ensure('installed') }
+
+        it {
+          is_expected.to contain_user('hyperglass-agent').with(
+            {
+              'ensure'         => 'present',
+              'managehome'     => true,
+              'purge_ssh_keys' => true,
+              'system'         => true,
+              'home'           => '/opt/hyperglass/hyperglass-agent',
+            }
+          )
+        }
+
+        it { is_expected.to contain_group('hyperglass-agent').with_ensure('present').with_system(true) }
+
+        it {
+          is_expected.to contain_file('/opt/hyperglass/hyperglass-agent').with(
+            {
+              'ensure' => 'directory',
+              'owner'  => 'hyperglass-agent',
+              'group'  => 'hyperglass-agent',
+              'mode'   => '0700',
+              'notify' => 'Systemd::Unit_file[hyperglass-agent.service]',
```

Comment:
Better to use the relationship matchers. https://github.com/rodjek/rspec-puppet#relationship-matchers

The tests will continue to work if the way the ralationships are defined are ever refactored.

Reply:
That's a great idea!

Reply:
Used those where the `.with_` and `.that_` were chained. Kept this as is because it makes it harder to read to use multiple formats for the same it block.